libs/@local/codegen/src/typescript.rs

```diff
@@ -0,0 +1,284 @@
+use oxc::{
+    allocator::Allocator,
+    ast::{AstBuilder, ast},
+    codegen::Codegen,
+    span::SPAN,
+};
+
+use crate::{
+    TypeCollection,
+    definitions::{Enum, EnumTagging, EnumVariant, Fields, Primitive, Type, TypeDefinition},
+};
+
+#[derive(Default)]
+#[non_exhaustive]
+pub struct TypeScriptGeneratorSettings {
+    allocator: Allocator,
+}
+
+#[expect(dead_code)]
+pub struct TypeScriptGenerator<'a> {
+    settings: &'a TypeScriptGeneratorSettings,
+    collection: TypeCollection,
+    ast: AstBuilder<'a>,
+}
+
+impl<'a> TypeScriptGenerator<'a> {
+    pub const fn new(
+        settings: &'a TypeScriptGeneratorSettings,
+        collection: TypeCollection,
+    ) -> Self {
+        Self {
+            settings,
+            collection,
+            ast: AstBuilder {
+                allocator: &settings.allocator,
+            },
+        }
+    }
+
+    #[must_use]
+    pub fn generate(&self, name: &str) -> String {
+        Codegen::new()
+            .build(
+                &self.ast.program(
+                    SPAN,
+                    ast::SourceType::d_ts(),
+                    "test",
```

Comment:
nit: we're not in a test environment here, so maybe a name like: `unnamed` or similar, might be more appropriate, or even a user supplied name

Reply:
```suggestion
                    "unnamed",
```